### ISSUE TITLE
Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory

### ISSUE THREAD

**Issue by tobiasehlert** [google-github-actions/auth#75]
### TL;DR

I've updated my workflow due to the deprecation of _service_account_key_ in the [google-github-actions/setup-gcloud](https://github.com/google-github-actions/setup-gcloud) action.

When running my workflow, the auth step fails after 50-55s resulting in the stacktrace shown below.

### Expected behavior

The authentication should not end up with a failure.

### Observed behavior

Auth step fails with following error..

```console
FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory

<--- Last few GCs --->

[409:0x3a26080]    49157 ms: Scavenge 2029.8 (2034.9) -> 2029.7 (2040.2) MB, 12.8 / 0.0 ms  (average mu = 0.245, current mu = 0.242) allocation failure 
[409:0x3a26080]    49175 ms: Scavenge 2032.9 (2040.2) -> 2033.2 (2040.9) MB, 16.1 / 0.0 ms  (average mu = 0.245, current mu = 0.242) allocation failure 
[409:0x3a26080]    49193 ms: Scavenge 2033.6 (2040.9) -> 2033.3 (2051.9) MB, 17.2 / 0.0 ms  (average mu = 0.245, current mu = 0.242) allocation failure 


<--- JS stacktrace --->

==== JS stack trace =========================================

    0: ExitFrame [pc: 0x1374fd9]
Security context: 0x0a48f43c08a1 <JSObject>
    1: fromBase64(aka fromBase64) [0x4e37389aad1] [/home/github/_work/_actions/google-github-actions/auth/v0.4.1/dist/index.js:~665] [pc=0x3540958c3e9d](this=0x00c3800004a9 <undefined>,0x3242bf7acf31 <Very long string[28855]>)
    2: parseServiceAccountKeyJSON [0x4e37389abb1] [/home/github/_work/_actions/google-github-actions/auth/v0.4.1/dist/index.js:1816] [byte...

 1: 0x9da7c0 node::Abort() [/home/github/externals/node12/bin/node]
 2: 0x9db976 node::OnFatalError(char const*, char const*) [/home/github/externals/node12/bin/node]
 3: 0xb39f1e v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [/home/github/externals/node12/bin/node]
 4: 0xb3a299 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [/home/github/externals/node12/bin/node]
 5: 0xce5635  [/home/github/externals/node12/bin/node]
 6: 0xce5cc6 v8::internal::Heap::RecomputeLimits(v8::internal::GarbageCollector) [/home/github/externals/node12/bin/node]
 7: 0xcf1b5a v8::internal::Heap::PerformGarbageCollection(v8::internal::GarbageCollector, v8::GCCallbackFlags) [/home/github/externals/node12/bin/node]
 8: 0xcf2a65 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [/home/github/externals/node12/bin/node]
 9: 0xcf410f v8::internal::Heap::HandleGCRequest() [/home/github/externals/node12/bin/node]
10: 0xca30f4 v8::internal::StackGuard::HandleInterrupts() [/home/github/externals/node12/bin/node]
11: 0xff1af7 v8::internal::Runtime_StackGuard(int, unsigned long*, v8::internal::Isolate*) [/home/github/externals/node12/bin/node]
12: 0x1374fd9  [/home/github/externals/node12/bin/node]
```

..which results in the workflow to fail.

### Action YAML

```yaml
- name: Authenticate to Google Cloud
        uses: google-github-actions/auth@v0.4.1
        with:
          credentials_json: ${{ secrets.GOOGLE_CREDENTIALS }}

      - name: Set up Cloud SDK
        uses: google-github-actions/setup-gcloud@v0.3.0
        with:
          project_id: ${{ secrets.GOOGLE_PROJECT_ID }}
          export_default_credentials: true
          version: 344.0.0
```

### Additional information

I'm running a self-hosted GitHub Runner with version 2.285.1 in a GKE cluster.

Could this maybe be somehow related to the #73 issue?

**Comment by sethvargo** [comment]
I think I fixed this in https://github.com/google-github-actions/auth/commit/8708e498da900968103a8e8cd9ad60479f849823#diff-b99a1f511f7ecc04caf1c201319b25852706b33bf09209372ea4db5178c9c5c7R740. Would you mind giving `auth@main` a try? We generally don't recommend running `@main`, but it would be good to test if that fix fixes your issue.

**Comment by tobiasehlert** [comment]
> I think I fixed this in [8708e49#diff-b99a1f511f7ecc04caf1c201319b25852706b33bf09209372ea4db5178c9c5c7R740](https://github.com/google-github-actions/auth/commit/8708e498da900968103a8e8cd9ad60479f849823#diff-b99a1f511f7ecc04caf1c201319b25852706b33bf09209372ea4db5178c9c5c7R740). Would you mind giving `auth@main` a try? We generally don't recommend running `@main`, but it would be good to test if that fix fixes your issue.

Switched to _main_ for testing and now the auth step is successful.
Now the next step is failing instead, which is the _google-github-actions/setup-gcloud@v0.3.0_ step.. which returns following error now:

```console
Error: google-github-actions/setup-gcloud failed with: Error parsing credentials: Unexpected end of JSON input
```

**Comment by bharathkkb** [comment]
@tobiasehlert that is possibly https://github.com/google-github-actions/auth/issues/73. Do you have `export_default_credentials: true` for `setup-gcloud` and if you do, could you try removing it. The `auth` action already does the export by default.

**Comment by tobiasehlert** [comment]
> @tobiasehlert that is possibly #73. Do you have `export_default_credentials: true` for `setup-gcloud` and if you do, could you try removing it. The `auth` action already does the export by default.

@bharathkkb removing _export_default_credentials_ from setup-gcloud and using _main_ on the auth step resolves my issues.

Btw, setting project-id should not be required when the project-id already is set by the auth step?
Feels like an unnecessary parameter to add to the setup-gcloud step.

**Comment by sethvargo** [comment]
I've released `auth@v0.4.2` to fix that regression on base64 decoding.

> Btw, setting project-id should not be required when the project-id already is set by the auth step?

Can you open an issue on the target repo (e.g. setup-gcloud) so we can track that please?

**Comment by tobiasehlert** [comment]
Thanks for your help @sethvargo and @bharathkkb 🎉
